flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-twitter/src/main/java/org/apache/flink/streaming/connectors/twitter/TwitterSource.java

```diff
@@ -133,13 +134,21 @@ protected OAuth1 authenticate() {
 	private Properties loadAuthenticationProperties() {
 		
 		Properties properties = new Properties();
+		InputStream input = null;
 		try {
-			InputStream input = new FileInputStream(authPath);
+			input = new FileInputStream(authPath);
 			properties.load(input);
-			input.close();
 		} catch (Exception e) {
 			throw new RuntimeException("Cannot open .properties file: " + authPath, e);
 		}
+		finally {
+			try {
```

Comment:
This try..catch does not look beautiful. But `input.close()` mandates it. Also `close()` will not be called if `FileInputStream` is not open.